### PR TITLE
OSIDB-2739: Remove component field from Flaw

### PR DESCRIPTION
**Pull request by MrMarble**
Removed the deprecated `component` field in favor of `components` from Flaw

Should be merged with https://github.com/RedHatProductSecurity/osim/pull/254

Closes OSIDB-2739

### CODE REVIEW

File: apps/bbsync/query.py [hunk 1]
```diff
@@ -178,7 +178,11 @@ def generate_summary(self):
         # 5) add trailing space delimiter in case of non-empty CVE IDs
         cve_ids = cve_ids + " " if cve_ids else cve_ids
 
-        component = self.flaw.component + ": " if self.flaw.component else ""
+        component = (
```
Comment (osoukup) [inline apps/bbsync/query.py:181]: nitpick: I think the variable name would now make more sense in plural

**Review by osoukup — APPROVED** [review]
LGTM

File: osidb/tests/endpoints/test_endpoints.py [hunk 2]
```diff
@@ -405,7 +405,7 @@ def test_flaw_components_create(self, auth_client, test_api_uri):
             "mitigation": "mitigation",
             "embargoed": False,
             "unembargo_dt": "2000-1-1T22:03:26.065Z",
-            "component": "curl",
+            "components": ["curl"],
```
Comment (jobselko) [inline osidb/tests/endpoints/test_endpoints.py:408]: This updates a test checking the deprecated functionality, so the test should be removed :). And you can also remove the whole `TestDeprecationHandling` class.

File: osidb/models.py
Comment (jobselko) [inline osidb/models.py:1]: `_validate_nonempty_component` should be changed (with its doctsring, error and test) to reflect the plural.

File: osidb/serializer.py
Comment (jobselko) [inline osidb/serializer.py:1]: Remove the following from `FlawSerializer`:
```
@extend_schema_serializer(
    deprecate_fields=[
        "component",
    ]
)
```

**Review by jobselko — CHANGES_REQUESTED** [review]

**Review by jobselko — APPROVED** [review]

**Review by JakubFrejlach — APPROVED** [review]

**Pull request merged**